content/pl/dev-prod-parity.md

```diff
@@ -0,0 +1,78 @@
+
+* Zmniejsz czas deploymentu: czas wdrożenia kodu napisanego przez developera powinien być mierzony w godzinach, a nawet w minutach.
+* Przenieś odpowiedzialność: developer piszący kod powinien być zaangażowany we wdrożenia aplikacji na produkcję.
+* Stosuj ten sam zestaw narzędzi : utrzymuj wszystkie środowiska w których działa aplikacja tak podobne jak to możliwe.
+
+Podsumowując w formie tabeli:
+
+<table>
+  <tr>
+    <th></th>
+    <th>Tradycyjna aplikacja</th>
+    <th>Aplikacja 12factor</th>
+  </tr>
+  <tr>
+    <th>Czas pomiędzy wdrożeniami</th>
+    <td>Tygodnie</td>
+    <td>Godziny</td>
+  </tr>
+  <tr>
+    <th>Tworzenie kodu vs wdrażanie kodu</th>
+    <td>Różne osoby</td>
+    <td>Te same osoby</td>
+  </tr>
+  <tr>
+    <th>Środowisko developerskie vs produkcyjne</th>
+    <td>Mocno różniące się</td>
+    <td>Jak najbardziej zbliżone</td>
+  </tr>
+</table>
+
+[Usługi wspierające](./backing-services), takie jak baza danych aplikacji, system kolejkowania czy cache są polem gdzie zachowanie podobieństw między wdrożeniami jest ważne. Wiele języków oferuje biblioteki, które upraszczają korzystanie z usług wspierających w tym *adaptery* do usług różnego typu. Kilka przykładów w tabeli poniżej:
+
+<table>
+  <tr>
+    <th>Typ</th>
+    <th>Język</th>
+    <th>Biblioteka</th>
+    <th>Adaptery</th>
+  </tr>
+  <tr>
+    <td>Baza danych</td>
+    <td>Ruby/Rails</td>
+    <td>ActiveRecord</td>
+    <td>MySQL, PostgreSQL, SQLite</td>
+  </tr>
+  <tr>
+    <td>Kolejka</td>
+    <td>Python/Django</td>
+    <td>Celery</td>
+    <td>RabbitMQ, Beanstalkd, Redis</td>
+  </tr>
+  <tr>
+    <td>Cache</td>
+    <td>Ruby/Rails</td>
+    <td>ActiveSupport::Cache</td>
+    <td>Pamięć, system plików, Memcached</td>
+  </tr>
+</table>
+
+Developerzy czasem wolą korzystać z lżejszych wersji różnych usług w swoim lokalnym środowisku, na produkcji w tym czasie używając ich w pełnych wersjach. Przykładem takiej sytuacji jest używanie lokalnie SQLite, a PostgreSQL na produkcji. Podobnie wygląda używanie na środowisku developerskim pamięci do cachowania, zamiast Memcached znajdującego się na produkcji.
+
+**Developer postępujący zgodnie zasadami 12factor opiera się pokusie używania usług różniących się pomiędzy środowiskami**, nawet wtedy gdy adaptery teoretycznie wraz z abstrakcją niwelują różnice w używaniu usług. Z powodu różnic pomiędzy usługami wspierającymi mogą pojawić się niezgodności, powodując, że kod, który działał i był testowany lokalnie lub na stagingu, przestanie funkcjonować na produkcji. Pojawianie się tego typu błędów negatywnie wpływa na proces ciągłego wdrażania aplikacji. Koszt czasu straconego na takich błędach i konsekwentnym psuciu się procesu ciągłego wdrażania jest niezwykle wysoki, biorąc pod uwagę gromadzenie się błędów w dłuższym okresie czasu.
```

Comment:
1. `adaptery teoretycznie z abstrakcją...` to nie brzmi dobrze. Coś w stylu `nawet wtedy, gdy adaptery teoretycznie ukrywają różnice w implementacji pod warstwą abstrakcji` brzmi lepiej.
2.